README.md

```diff
@@ -230,39 +225,149 @@ module.exports = {
 };
 ```
 
-Migration files should be located in the same directory, according to the info you gave to the `Umzug` constructor.
+Migration files can be located anywhere - they will typically be loaded according to a glob pattern provided to the `Umzug` constructor.
 
 #### Direct migrations list
 
-You can also specify directly a list of migrations to the `Umzug` constructor. We recommend the usage of the `Umzug.migrationsList()` function
-as bellow:
+You can also specify directly a list of migrations to the `Umzug` constructor:
 
 ```js
-const { Umzug, migrationsList } = require('umzug');
+const { Umzug } = require('umzug');
 
 const umzug = new Umzug({
-  migrations: migrationsList(
-    [
-      {
-        // the name of the migration is mandatory
-        name: '00-first-migration',
-        async up(queryInterface) { /* ... */ },
-        async down(queryInterface) { /* ... */ }
+  migrations: [
+    {
+      // the name of the migration is mandatory
+      name: '00-first-migration',
+      async up({ context }) { /* ... */ },
+      async down({ context }) { /* ... */ }
+    },
+    {
+      name: '01-foo-bar-migration',
+      async up({ context }) { /* ... */ },
+      async down({ context }) { /* ... */ }
+    }
+  ],
+  context: sequelize.getQueryInterface(),
+  logger: console,
+});
+```
+
+To load migrations in another format, you can use the `resolve` function:
+
+```js
+const { Umzug } = require('umzug');
+const fs = require('fs')
+
+const umzug = new Umzug({
+  migrations: {
+    glob: 'migrations/*.up.sql',
+    resolve: ({ name, path, context }) => ({
+      name,
+      up: async () => {
+        const sql = fs.readFileSync(path).toString()
+        return context.sequelize.query(sql)
       },
-      {
-        name: '01-foo-bar-migration',
-        async up(queryInterface) { /* ... */ },
-        async down(queryInterface) { /* ... */ }
+      down: async (({ name, path, context })) => {
+        const sql = fs.readFileSync(path.replace('.up.sql', '.down.sql')).toString()
+        return context.sequelize.query(sql)
       }
-    ],
-    // an optional list of parameters that will be sent to the `up` and `down` functions
-    [
-      sequelize.getQueryInterface()
-    ]
-  )
+    })
+  },
+  context: sequelize.getQueryInterface(),
+  logger: console,
 });
 ```
 
+You can support mixed migration file types, and use umzug's default resolver for javascript/typescript:
+
+```js
+const { Umzug } = require('umzug');
+const fs = require('fs')
+
+const umzug = new Umzug({
+  migrations: {
+    glob: 'migrations/*.{js,ts,up.sql}',
+    resolve: (params) => {
+      if (!params.path.endsWith('.sql')) {
+        return Umzug.defaultResolver(params)
+      }
+      return {
+        name: params.name,
+        up: async () => {
+          const sql = fs.readFileSync(params.path).toString()
+          return params.context.sequelize.query(sql)
+        },
+        down: async (params) => {
```

Comment:
```suggestion
        down: async () => {
          // Get the corresponding `.down.sql` file to undo this migration
```

Reply:
I think this is not resolved yet

Reply:
Ah mixed this up with the one above, good catch!